integration/tsic/tsic.go

```diff
@@ -555,6 +575,62 @@ func (t *TailscaleInContainer) Ping(hostnameOrIP string, opts ...PingOption) err
 	})
 }
 
+// PingViaDERP executes the Tailscale ping command and pings a hostname
+// or IP via the DERP network (i.e., not a direct connection). It accepts a series of DERPPingOption.
+// TODO(kradalby): Make multiping, go routine magic.
+func (t *TailscaleInContainer) PingViaDERP(hostnameOrIP string, opts ...PingOption) error {
+	args := pingArgs{
+		timeout: time.Second,
+		count:   defaultPingCount,
+	}
+
+	for _, opt := range opts {
+		opt(&args)
+	}
+
+	command := []string{
+		"tailscale", "ping",
+		fmt.Sprintf("--timeout=%s", args.timeout),
+		fmt.Sprintf("--c=%d", args.count),
+		"--until-direct=false",
+	}
+
+	command = append(command, hostnameOrIP)
+
+	return t.pool.Retry(func() error {
+		result, _, err := t.Execute(
+			command,
+			dockertestutil.ExecuteCommandTimeout(
+				time.Duration(int64(args.timeout)*int64(args.count)),
+			),
+		)
+		if err != nil {
+			fmt.Printf(
+				"failed to run ping command from %s to %s, err: %s",
+				t.Hostname(),
+				hostnameOrIP,
+				err,
+			)
+
+			return err
+		}
+
+		if strings.Contains(result, "is local") {
+			return nil
+		}
+
+		if !strings.Contains(result, "pong") {
+			return backoff.Permanent(errTailscalePingFailed)
+		}
+
+		if !strings.Contains(result, "via DERP") {
```

Comment:
Taking a second look at this function, it seems unnecessary, we can achieve the same by using `WithPingUntilDirect` and wrapping this if check in `if !args.direct` in the original function

Reply:
I agree :)